### ISSUE TITLE
Support inline metadata for objects

### ISSUE THREAD

**Issue by cdmgtri** [niemopen/niem-naming-design-rules#8]
Migrated from NIEM/NIEM-NDR#112 

"Make the default behavior correct." -- Christina

#todo Christina and Tom Carlson more examples, esp Brian Handspicker stuff

#resolved kill off `relationshipMetadata`

Some people say metadata is hard to use.  Could restructure things a bit to make it easier.

- Combine the concepts of `structures:metadata` and `structures:relationshipMetadata`

Dr. Scott: Which RDF one is the one that we keep? `relationshipMetadata` takes you out of tuple-space

- Allow users to extend or augment a common metadata object to avoid references to multiple objects

- Plug in an `structures:Metadata` element containing only an augmentation point and add it to `structures:ObjectType`?

  - Get rid of the appliesToTypes and appliesToProperties as they are rarely used
  - Core, domains, and IEPDs could augment as needed
  - IEPDs could fill in metadata inline within objects for simple cases, or still use the standard `id/ref` attributes to reference reusable metadata blocks defined outside of the object
    - Core, domains, and IEPDs could still extend `structures:MetadataType` and add it explicitly to types if they need the current `appliesTo` capability

**Current**

```xml
<nc:Person>
  <nc:PersonBirthDate structures:metadata="m1">
    <nc:Date>1945-12-01</nc:Date>
  </nc:PersonBirthDate>
  <nc:PersonName structures:metadata="m1 m2">
    <nc:PersonFullName>John Doe</nc:PersonFullName>
  </nc:PersonName>
</nc:Person>

<nc:Metadata structures:id="m1">
  <nc:SourceText>Adam Barber</nc:SourceText>
  <nc:ReportedDate>
    <nc:Date>2005-04-26</nc:Date>
  </nc:ReportedDate>
</nc:Metadata>

<j:Metadata structures:id="m2">
  <j:CriminalInformationIndicator>true</j:CriminalInformationIndicator>
</j:Metadata>
```

**Proposed - Simple use case (inline)**

```xml
<nc:Person>

  <!-- Inline metadata -->
  <structures:Metadata>
    <nc:MetadataAugmentation>
      <nc:SourceText>Adam Barber</nc:SourceText>
      <nc:ReportedDate>
        <nc:Date>2005-04-26</nc:Date>
      </nc:ReportedDate>
    </nc:MetadataAugmentation>
    <j:MetadataAugmentation>
      <j:CriminalInformationIndicator>true</j:CriminalInformationIndicator>
    </j:MetadataAugmentation>
  </structures:Metadata>

  <nc:PersonBirthDate>
    <nc:Date>1945-12-01</nc:Date>
  </nc:PersonBirthDate>
  <nc:PersonName>
    <nc:PersonFullName>John Doe</nc:PersonFullName>
  </nc:PersonName>

</nc:Person>
```

**Proposed - Advanced use case (reference)**

```xml
<nc:Person>
  <!-- Metadata reference, similar to how other objects in NIEM can be referenced -->
  <structures:Metadata structures:ref="m1" xsi:nil="true"/>
  <nc:PersonBirthDate>
    <nc:Date>1945-12-01</nc:Date>
  </nc:PersonBirthDate>
  <nc:PersonName>
    <nc:PersonFullName>John Doe</nc:PersonFullName>
  </nc:PersonName>
</nc:Person>

<structures:Metadata id="m1">
  <nc:MetadataAugmentation>
    <nc:SourceText>Adam Barber</nc:SourceText>
    <nc:ReportedDate>
      <nc:Date>2005-04-26</nc:Date>
    </nc:ReportedDate>
  </nc:MetadataAugmentation>
  <j:MetadataAugmentation>
    <j:CriminalInformationIndicator>true</j:CriminalInformationIndicator>
  </j:MetadataAugmentation>
</structures:Metadata>
```

Questions:

- Make metadata properties direct substitutions for `structures:AugmentationPoint` rather than bundling them into augmentation containers?
- How does this affect CUI metadata?  More obvious to be able to put metadata inline where they apply?  Would losing the metadata containers make it more difficult to process?

```xml
<nc:Person>

  <!-- Removed Core and Justice augmentation element containers -->
  <structures:Metadata>
    <nc:SourceText>Adam Barber</nc:SourceText>
    <nc:ReportedDate>
      <nc:Date>2005-04-26</nc:Date>
    </nc:ReportedDate>
    <j:CriminalInformationIndicator>true</j:CriminalInformationIndicator>
  </structures:Metadata>

  <nc:PersonBirthDate>
    <nc:Date>1945-12-01</nc:Date>
  </nc:PersonBirthDate>
  <nc:PersonName>
    <nc:PersonFullName>John Doe</nc:PersonFullName>
  </nc:PersonName>

</nc:Person>
```

**Comment by cdmgtri** [comment]
Harmonization Subcommittee feedback:

Add the following to make metadata easier to use in well-known places.  Message designers can still add, extend, or augment types on their own to place metadata exactly where they need it to appear.

- Add new attribute augmentation `cbrn:totalDoseMetadataRef` to `cbrn:TotalDoseuSvType` 
- Add new attribute augmentation `cbrn:totalExposureMetadataRef` to `cbrn:TotalExposuremRType`
- Add `cui:DocumentMarkingMetadata` to CUI augmentations of `nc:DocumentType`, `nc:ReportType`, and `nc:MessageType`
- Add `mo:ImplementationSpecificSettingMetadata` to type `mo:SettingType`
- Add `mo:MinimumEssentialMetadata` to MilOps augmentations of `nc:DocumentType`, `nc:ReportType`, and `nc:MessageType`
- Add `mo:UncertaintyMetadata` to MilOps augmentation of `nc:MeasureType`
- Add `scr:PersonMetadata` to Screening augmentation of nc:PersonType

Consider whether we can add a new attribute augmentation `cui:portionMarkingMetadataRef` to `nc:TextType` in the reference model.  CUI is an auxiliary namespace.  Is it allowed to be referenced in Core?

It was discussed whether to add `nc:Metadata` to a Core augmentation of `structures:ObjectType` but rejected because if it is available everywhere, it's not easy to tell in a subset exactly which places it should be used.

**Comment by cdmgtri** [comment]
Model updates merged into the dev branch for 6.0.

**Comment by iamdrscott** [comment]
Handled in PS01 and NDR6
